configurator/src/capture.rs

```diff
@@ -218,18 +226,33 @@ impl Capturer {
         Ok(())
     }
 
-    fn distance_2d((x1, y1): (usize, usize), (x2, y2): (usize, usize)) -> f64 {
-        ((x1 as f64 - x2 as f64).powi(2) + (y1 as f64 - y2 as f64).powi(2)).sqrt()
+    fn distance_2d((x1, y1): &(usize, usize), (x2, y2): &(usize, usize)) -> f64 {
+        ((*x1 as f64 - *x2 as f64).powi(2) + (*y1 as f64 - *y2 as f64).powi(2)).sqrt()
     }
 
-    fn remove_outliers_by_2d_distance(
-        raw_points: Vec<WithConfidence<(usize, usize)>>,
-    ) -> Vec<WithConfidence<(usize, usize)>> {
+    fn distance_3d(a: &Vector3<f64>, b: &Vector3<f64>) -> f64 {
+        a.metric_distance(b)
+    }
+
+    pub fn reduce_confidence_of_outliers_by_3d_distance(
+        raw_points: Vec<WithConfidence<Vector3<f64>>>,
+    ) -> Vec<WithConfidence<Vector3<f64>>> {
+        Self::reduce_confidence_of_outliers_by_distance(raw_points, Self::distance_3d)
+    }
```

Comment:
Wouldn't it be better to just call the original function directly with:

```rust
Capturer::reduce_confidence_of_outliers_by_distance(raw_points, |(a, b)| a.metric_distance(b));
```

Rather than creating 2 additional functions that add nothing but naming?

Reply:
I thought that making the whole templated method public was a bit "leaky", especially creating the distance measuring function on the call site, but if that's your preference then sure

Reply:
If anything I think the whole capture process logic that is now in main should be moved to capturer. But that's out of scope here